src/tests/profiler/native/eventpipeprofiler/eventpipewritingprofiler.cpp

```diff
@@ -188,6 +211,34 @@ HRESULT STDMETHODCALLTYPE EventPipeWritingProfiler::JITCachedFunctionSearchFinis
     return S_OK;
 }
 
+void EventPipeWritingProfiler::ProviderCallback(
+    const UINT8 *source_id,
+    UINT32 is_enabled,
+    UINT8 level,
+    UINT64 match_any_keywords,
+    UINT64 match_all_keywords,
+    COR_PRF_FILTER_DATA *filter_data,
+    void *callback_data)
+{
+    if (is_enabled)
+    {
+        _sawEnable = true;
+    }
+    else
+    {
+        _sawDisable = true;
+    }
+
+    // The diagnostics client library sets keywords to 0xF00000000000 by default
+    if (match_any_keywords != 0xF00000000000
```

Comment:
Does this mean keywords are also set when delivering the disable notification? That seems surprising although not necessarily problematic.

Is there particular behavior we are promising when a provider has been enabled more than once, or when we are disabling one session for a provider but other sessions are still running? Historically these behaviors have been poorly specified and it might be a lot more awkward to change the behavior in the future vs. ensuring it is what we want now.

Reply:
The current behavior is `is_enabled` will be true if any session has the provider active, and keywords will be whatever the combined keywords for all sessions are (on enable and disable).

This behavior matches ETW as far as I can tell